SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/WorkerRegion.cpp

```diff
@@ -24,16 +27,54 @@ AWorkerRegion::AWorkerRegion(const FObjectInitializer& ObjectInitializer)
 	SetRootComponent(Mesh);
 }
 
-void AWorkerRegion::Init(UMaterial* Material, const FColor& Color, const float Opacity, const FBox2D& Extents, const float Height,
-						 const float VerticalScale)
+void AWorkerRegion::Init(UMaterial* BackgroundMaterial, UMaterial* InCombinedMaterial, UFont* InWorkerInfoFont, const FColor& Color,
+						 const float Opacity, const FBox2D& Extents, const float Height, const float VerticalScale,
+						 const FString& InWorkerInfo, const bool bInEditor)
 {
+	// Background translucent coloured worker material
+	BackgroundMaterialInstance = UMaterialInstanceDynamic::Create(BackgroundMaterial, nullptr);
 	SetHeight(Height);
 
-	MaterialInstance = UMaterialInstanceDynamic::Create(Material, nullptr);
-	Mesh->SetMaterial(0, MaterialInstance);
+	if (bInEditor)
+	{
+		// In editor, setup the basic boundary material
+		Mesh->SetMaterial(0, BackgroundMaterialInstance);
+	}
+	else
+	{
+		// At runtime, setup the dynamic boundary material
+		CombinedMaterial = InCombinedMaterial;
+		WorkerInfoFont = InWorkerInfoFont;
+		WorkerInfo = InWorkerInfo;
+		CombinedMaterialInstance = UMaterialInstanceDynamic::Create(CombinedMaterial, nullptr);
+		Mesh->SetMaterial(0, CombinedMaterialInstance);
+		CanvasRenderTarget = UCanvasRenderTarget2D::CreateCanvasRenderTarget2D(this, UCanvasRenderTarget2D::StaticClass(), 1024, 1024);
+		CanvasRenderTarget->OnCanvasRenderTargetUpdate.AddDynamic(this, &AWorkerRegion::DrawToCanvasRenderTarget);
+	}
+
 	SetOpacity(Opacity);
 	SetColor(Color);
 	SetPositionAndScale(Extents, VerticalScale);
+
+	if (!bInEditor)
+	{
+		// At runtime, calls DrawToCanvasRenderTarget to render the dynamic boundary material
+		CanvasRenderTarget->UpdateResource();
```

Comment:
How is this texture updated when running in editor?

Reply:
This approach to dynamically render the text on the material via a Canvas only works at runtime as the UCanvasRenderTarget2D->OnCanvasRenderTargetUpdate() callback is only triggered at runtime.